CMakeLists.txt

```diff
@@ -2,6 +2,15 @@ cmake_minimum_required(VERSION 3.25...4.0 FATAL_ERROR)
 
 project(isle LANGUAGES CXX C VERSION 0.1)
 
+if (EMSCRIPTEN)
+  set(CMAKE_EXECUTABLE_SUFFIX ".html")
+  add_compile_options("$<$<PLATFORM_ID:Emscripten>:-pthread>")
+  add_link_options("$<$<PLATFORM_ID:Emscripten>:-sALLOW_MEMORY_GROWTH=1>")
+  add_link_options("$<$<PLATFORM_ID:Emscripten>:-sMAXIMUM_MEMORY=2gb>")
```

Comment:
I don't think the generator expressions are needed, since you already add them conditionally when targeting `EMSCRIPTEN`.